mdoc-sbt/src/main/scala/mdoc/MdocPlugin.scala

```diff
@@ -106,21 +112,52 @@ object MdocPlugin extends AutoPlugin {
         mdocInternalVariables.value.foreach { case (key, value) =>
           props.put(key, value)
         }
+        def getJars(mid: ModuleID) = {
+
+          val depRes = dependencyResolution.in(update).value
+          val updc = updateConfiguration.in(update).value
+          val uwconfig = unresolvedWarningConfiguration.in(update).value
+          val modDescr = depRes.wrapDependencyInModule(mid)
+
+          depRes
+            .update(
+              modDescr,
+              updc,
+              uwconfig,
+              streams.value.log
+            )
+            .map(_.allFiles)
+            .fold(uw => throw uw.resolveException, identity)
+        }
+
+        val binaryVersion = scalaBinaryVersion.value
+        val log = streams.value.log
+        val libraries = mdocJSLibraries.value.map(_.data)
+        val workerClasspathOverride = mdocJSWorkerClasspath.value
+
         mdocJSCompileOptions.value.foreach { options =>
-          props.put(
-            s"js-scalac-options",
-            options.options.mkString(" ")
-          )
-          props.put(
-            s"js-classpath",
-            options.classpath.mkString(File.pathSeparator)
-          )
-          options.moduleKind.foreach { moduleKind => props.put(s"js-module-kind", moduleKind) }
+          val sjsVersion = detectScalaJSVersion
+
+          val linkerDependency = binaryVersion match {
+            case "3" => "org.scala-js" % "scalajs-linker_2.13" % sjsVersion
+            case other => "org.scala-js" % s"scalajs-linker_$other" % sjsVersion
+          }
+
+          val mdocJSDependency = binaryVersion match {
+            case "3" => "org.scalameta" % "mdoc-js-worker_3" % BuildInfo.version
+            case other => "org.scalameta" % s"mdoc-js-worker_$other" % BuildInfo.version
+          }
+
+          val workerClasspath = workerClasspathOverride.getOrElse(getJars(mdocJSDependency))
+
+          MdocJSConfiguration(
+            scalacOptions = options.options,
+            compileClasspath = options.classpath,
+            linkerClassPath = getJars(linkerDependency) ++ workerClasspath,
+            moduleKind = options.moduleKind,
+            jsLibraries = libraries
+          ).writeTo(props)
```

Comment:
This is a strictly unnecessary refactor, but it's a small step towards my goal of weaning off mdoc's build of loading MdocPlugin at build time.

Main issue is that it forces 2.12 on all the modules that are used to build the site
It also causes some weird issues when compiling the build, i.e. I currently can't resolve the deprecation warnings because the build just fails to compile when `++2.12` is used

Reply:
sbt on 2.13 would be so nice.